### PR TITLE
feat: Added esbuild integration tests to sam sync

### PR DESCRIPTION
**Pull request by lucashuy**
#### Mandatory Checklist
**PRs will only be reviewed after checklist is complete**

- [x] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [x] Write design document if needed ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [x] Write/update unit tests
- [x] Write/update integration tests
- [x] Write/update functional tests if needed
- [x] `make pr` passes
- [x] `make update-reproducible-reqs` if dependencies were changed
- [x] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

File: tests/integration/sync/test_sync_code.py [hunk 1]
```diff
@@ -618,3 +621,59 @@ def test_sync_code_nested_getattr_layer(self):
...
+
+        # Run code sync
+        sync_command_list = self.get_sync_command_list(
+            template_file=TestSyncCodeBase.template_path,
+            code=True,
+            watch=False,
+            resource_list=["AWS::Serverless::Function"],
+            dependency_layer=self.dependency_layer,
+            stack_name=TestSyncCodeBase.stack_name,
+            parameter_overrides="Parameter=Clarity",
```
Comment (sriram-mv) [inline tests/integration/sync/test_sync_code.py:655]: Do you need paramater overrides, image_repository, tags, kms_key_id you could get rid of a few params here.

File: tests/integration/sync/test_sync_infra.py [hunk 2]
```diff
@@ -403,3 +404,42 @@ def _run_sync_and_validate_lambda_call(self, dependency_layer: bool, template_pa
...
+class TestSyncInfraWithEsbuild(SyncIntegBase):
+    @parameterized.expand(["code/before/template-esbuild.yaml"])
+    def test_sync_infra_esbuild(self, template_file):
+        set_experimental(ExperimentalFlag.Esbuild)
+
+        template_path = str(self.test_data_path.joinpath(template_file))
+        stack_name = self._method_to_stack_name(self.id())
+        self.stacks.append({"name": stack_name})
+
+        self._run_sync_and_validate_lambda_call(self.dependency_layer, template_path, stack_name)
```
Comment (sriram-mv) [inline tests/integration/sync/test_sync_infra.py:420]: why are we running the same twice? are we checking for behavior?

File: tests/integration/testdata/sync/code/after/esbuild_function/app.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+let response;
+
+exports.lambdaHandler = async (event, context) => {
+    try {
+        response = {
+            'statusCode': 200,
+            'body': JSON.stringify({
+                message: 'Hello world!',
+                extra_message: "banana"
```
Comment (sriram-mv) [inline tests/integration/testdata/sync/code/after/esbuild_function/app.js:11]: Is this a minions reference haha :)

**Review by sriram-mv — CHANGES_REQUESTED** [review]

File: tests/integration/sync/test_sync_infra.py (hunk 2 above)
Comment (lucashuy) [reply tests/integration/sync/test_sync_infra.py:420]: Remnant from reusing an existing test, will remove.

**Review by qingchm — APPROVED** [review]

**Review by mildaniel — APPROVED** [review]

**Pull request merged**